src/value.jl

```diff
@@ -35,6 +35,9 @@ catvalue_type(::Type{T}, ::Type{R}) where {T, R} =
     CategoricalValue{T, R}
 catvalue_type(::Type{<:AbstractString}, ::Type{R}) where {R} =
     CategoricalString{R}
+# default to CategoricalString in degenerated case (all nulls)
+catvalue_type(::Type{Null}, ::Type{R}) where {R} =
```

Comment:
This sounds a bit weird. Doesn't it work if you return `CategoricalValue{Union{}, R}`? That's really the correct type, since we know it won't contain anything.